src/main/java/nextstep/session/service/SessionStudentService.java

```diff
@@ -0,0 +1,42 @@
+package nextstep.session.service;
+
+import nextstep.session.domain.Session;
+import nextstep.session.domain.student.SessionStudent;
+import nextstep.session.domain.student.SessionStudentRepository;
+import nextstep.session.domain.student.SessionStudentStatus;
+import nextstep.session.domain.student.SessionStudents;
+import nextstep.users.domain.NsUser;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+public class SessionStudentService {
+    private SessionStudentRepository studentRepository;
+
+    public SessionStudentService(SessionStudentRepository studentRepository) {
+        this.studentRepository = studentRepository;
+    }
+
+    @Transactional
+    public Long enrollStudent(Session session, NsUser nsUser) {
+        SessionStudent student = session.enrollStudent(nsUser);
+        return studentRepository.enrollStudent(session.getId(), student.getId());
+    }
+
+    @Transactional
+    public int approveStudent(SessionStudent student) {
+        return studentRepository.changeStudentStatus(student.getId(), SessionStudentStatus.APPROVE);
```

Comment:
이와 같이 구현할 경우 칼럼 값 변경에 따라 update를 위한 쿼리문과 repository api가 많아진다.
SessionStudent에서 변경 가능한 모든 값을 update할 수 있는 api를 하나 만들고 재사용하는 방식으로 구현하는 것은 어떨까?

이와 같이 구현할 경우 다음과 같은 구현이 가능해 지지 않을까?
```suggestion
        // 수강 승인
        student.approve();
        return studentRepository.save(student);
```
```suggestion
        // 수강 취소
        student.cancel();
        return studentRepository.save(student);
```